bftengine/src/bftengine/InternalBFTClient.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "PrimitiveTypes.hpp"
+#include "MsgsCommunicator.hpp"
+
+namespace bftEngine {
+namespace impl {
+
+class InternalBFTClient {
+ public:
+  InternalBFTClient(const int& id, const NodeIdType& nonInternalNum, MsgsCommunicator* msgComm);
+  NodeIdType getClientId();
+  void sendRquest(uint8_t flags, uint32_t requestLength, const char* request, const std::string& cid);
+
+ private:
+  uint32_t repID_{};
+  NodeIdType startIdForInternalClient_{};
+  MsgsCommunicator* msgComm_{nullptr};
```

Comment:
It's not the owner of the resource. 
For users of resources, I think it's valid, safe and reduces overhead to use raw pointers. 
what do you think?

Reply:
I think it would be better to use a shared_ptr, as it will guarantee proper destruction. In theory the raw pointer may not be valid because it doesn't know the lifetime of the original pointer. If you could guarantee that the pointer would live at least as long as the owner it would be safe. But in that case you should just us a reference, since the pointer is already created.

The common case for using a pointer when not owning is when just passing to a local function in the call chain that doesn't store the pointer.